### PR TITLE
use c++ style cast

### PR DESCRIPTION
**Pull request by mehagar**
I want to eventually enable the cpplint rule to enable prevent using c-style casts. This was about 1/5th of the C-style casts here.

### CODE REVIEW

File: source/core/Date.cpp [hunk 1]
```diff
@@ -301,7 +301,7 @@ namespace Vireo {
             gmtime_r(&time, &tm);
         else
             localtime_r(&time, &tm);
-        _timeZoneOffset = int(isUTC ? 0 : tm.tm_gmtoff);
+        _timeZoneOffset = static_cast<int>(isUTC ? 0 : tm.tm_gmtoff);
```
Comment (spathiwa) [inline source/core/Date.cpp]: Nothing wrong with a constructor cast here.

File: source/core/Date.cpp [hunk 2]
```diff
@@ -349,7 +349,7 @@ namespace Vireo {
         struct tm tm;
         time_t timeVal = utcTime - kStdDT1970re1904;
         localtime_r(&timeVal, &tm);
-        _systemLocaleTimeZone = int(tm.tm_gmtoff);
+        _systemLocaleTimeZone = static_cast<int>(tm.tm_gmtoff);
```
Comment (spathiwa) [inline source/core/Date.cpp]: and here.

File: source/core/Date.cpp (hunk 1 above)
Comment (cglzaguilar) [reply source/core/Date.cpp]: Yes, but let's use the static_cast to keep the code using only one kind of casting.

File: source/core/Date.cpp (hunk 2 above)
Comment (cglzaguilar) [reply source/core/Date.cpp]: Yes, but let's use the static_cast to keep the code using only one kind of casting.

**Pull request merged**